spec/unit/settings/http_extra_headers_spec.rb

```diff
@@ -0,0 +1,67 @@
+require 'spec_helper'
+
+require 'puppet/settings'
+require 'puppet/settings/http_extra_headers_setting'
+
+describe Puppet::Settings::HttpExtraHeadersSetting do
+  subject { described_class.new(:settings => double('settings'), :desc => "test") }
+
+  it "is of type :http_extra_headers" do
+    expect(subject.type).to eq :http_extra_headers
+  end
+
+  describe "munging the value" do
+    let(:final_value) { [['header1', 'foo'], ['header2', 'bar']] }
+
+    describe "when given a string" do
+      it "splits multiple values into an array" do
+        expect(subject.munge("header1:foo,header2:bar")).to match_array(final_value)
+      end
+
+      it "strips whitespace between elements" do
+        expect(subject.munge("header1:foo , header2:bar")).to match_array(final_value)
+      end
+
+      it "creates an array when one item is given" do
+        expect(subject.munge("header1:foo")).to match_array([['header1', 'foo']])
+      end
+    end
+
+    describe "when given an array of strings" do
+      it "returns an array of arrays" do
+        expect(subject.munge(['header1:foo', 'header2:bar'])).to match_array(final_value)
+      end
+    end
+
+    describe "when given an array of arrays" do
+      it "returns an array of arrays" do
+        expect(subject.munge([['header1', 'foo'], ['header2', 'bar']])).to match_array(final_value)
+      end
+    end
+
+    describe "when given a hash" do
+      it "returns the hash" do
+        expect(subject.munge({'header1' => 'foo', 'header2' => 'bar'})).to match_array(final_value)
+      end
+    end
+
+    #TODO should we accept this? should we raise an error?
+    describe 'when given headers with no value'
```

Comment:
Open question on how we want to handle this situation

Reply:
According to https://tools.ietf.org/html/rfc7230#section-3.2

```
     header-field   = field-name ":" OWS field-value OWS

     field-name     = token
     field-value    = *( field-content / obs-fold )
     field-content  = field-vchar [ 1*( SP / HTAB ) field-vchar ]
     field-vchar    = VCHAR / obs-text
```

So each `field-value` consists of one or more `VCHAR` (visible US-ASCII), so this case should be an error.